src/main/java/com/google/devtools/build/lib/runtime/InstrumentationOutputFactory.java

```diff
@@ -118,13 +121,16 @@ public InstrumentationOutput createInstrumentationOutput(
 
     // Since PathFragmentConverter for flag value replaces prefixed `~/` with user's home path, the
     // destination is either an absolute path or a path relative to output_base/workspace.
+    Path path =
+        switch (destinationRelativeTo) {
+          case OUTPUT_BASE -> env.getOutputBase().getRelative(destination);
+          case WORKSPACE_OR_HOME -> env.getWorkspace().getRelative(destination);
+          case WORKING_DIR -> env.getWorkingDirectory().getRelative(destination);
```

Comment:
As the author of `InstrumentationOutput`, I want to sincerely thank you for adopting `InstrumentationOutput`.

However, please note that `env.getWorkingDirectory()` is not "getting the directory from which the user ran Bazel". From [its implementation and JavaDoc](https://github.com/keithl-stripe/bazel/blob/6da8086ebc721e998023831cc76b7b6e248122bc/src/main/java/com/google/devtools/build/lib/analysis/BlazeDirectories.java#L121-L123), this method returns "working directory of the server", which could be identical to workspace, in most cases I think. I apologize for this confusion. And this is painful for us as well, which makes us making the initial assumption to conflate workspace and working directory.

As of now, we are researching about fixing this flawed assumption. So it is totally fine to opt out of adopting `InstrumentationOutput` in this change. I think we can do the switch after we know how to sort out this ball of mud.

Reply:
Ok, thanks for the explanation! Note: you linked to `BlazeDirectories.getWorkingDirectory`, which you described accurately; but I was not calling that; I was calling `CommandEnvironment.getWorkingDirectory()`, which _is_ the client working directory :)

https://github.com/bazelbuild/bazel/blob/72a9d491b3cd9a5156a701ae1f0b612668b91df7/src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java#L653-L662

Reply:
Sorry, I misread on this. Then this looks good to me.